homeassistant/components/airthings_ble/config_flow.py

```diff
@@ -0,0 +1,169 @@
+        except BleakError as err:
+            _LOGGER.error(
+                "Error connecting to and getting data from %s: %s",
+                discovery_info.address,
+                err,
+            )
+            raise AirthingsDeviceUpdateError("Failed getting device data") from err
+        except Exception as err:
+            _LOGGER.error(
+                "Unknown error occurred from %s: %s", discovery_info.address, err
+            )
+            raise err
+        return data
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfo
+    ) -> FlowResult:
+        """Handle the bluetooth discovery step."""
+        _LOGGER.debug("Discovered BT device: %s", discovery_info)
+        await self.async_set_unique_id(discovery_info.address)
+        self._abort_if_unique_id_configured()
+
+        try:
+            device = await self._get_device_data(discovery_info)
+        except AirthingsDeviceUpdateError:
+            return self.async_abort(reason="cannot_connect")
+        except Exception:  # pylint: disable=broad-except
+            return self.async_abort(reason="unknown")
+
+        name = get_name(device)
+        self.context["title_placeholders"] = {"name": name}
+        self._discovered_device = Discovery(name, discovery_info, device)
+
+        return await self.async_step_bluetooth_confirm()
+
+    async def async_step_bluetooth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm discovery."""
+        if user_input is not None:
+            return self.async_create_entry(
+                title=self.context["title_placeholders"]["name"], data={}
+            )
+
+        self._set_confirm_only()
+        return self.async_show_form(
+            step_id="bluetooth_confirm",
+            description_placeholders=self.context["title_placeholders"],
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the user step to pick discovered device."""
+        if user_input is not None:
+            address = user_input[CONF_ADDRESS]
+            await self.async_set_unique_id(address, raise_on_progress=False)
+            self._abort_if_unique_id_configured()
+            discovery = self._discovered_devices[address]
+
+            self.context["title_placeholders"] = {
+                "name": discovery.name,
+            }
+
+            self._discovered_device = discovery
+
+            return self.async_create_entry(title=discovery.name, data={})
+
+        current_addresses = self._async_current_ids()
+        for discovery_info in async_discovered_service_info(self.hass):
+            address = discovery_info.address
+            if address in current_addresses or address in self._discovered_devices:
+                continue
+
+            if 820 not in discovery_info.manufacturer_data:
```

Comment:
Maybe make `820` a constant?  

Is there something else you can check as well.  Many devices abuse this field and put random data in it so you might pick up something close that isn't actually using this field correctly

Reply:
Since there is basically no more data, I am a bit unsure.

We're connecting to it  for name + S/N anyways so could check:
- Is connectable even?
- Do we have any sensor data?

Might this improve accuracy?

Reply:
Have you got a capture of a "real" ServiceInfo? Is it just a blank manufacturer info? Any service uuids?

Reply:
This is what I've got. I'm a bit unsure what service uuids are, and if they're the same across Airthings.

```
BluetoothServiceInfoBleak(
  name='mac', 
  address='mac, 
  rssi=-59, 
  manufacturer_data={820: b'\xe4/\xa5\xae\t\x00'}, 
  service_data={}, 
  service_uuids=['b42e1c08-ade7-11e4-89d3-123b93f75cba'], 
  source='hci0', 
  device=mac:mac, 
  advertisement=AdvertisementData(manufacturer_data={820: b'\xe4/\xa5\xae\t\x00'}, 
  service_uuids=['b42e1c08-ade7-11e4-89d3-123b93f75cba']), 
  connectable=True, 
  time=360.593692623
)
```

Reply:
I can see other airthings devices uses that uuid so I'm confident it's not specific to you at least. And it doesn't seem to be a generic one either.

It's probably better to miss some devices than "discover" a completely unrelated sensor that's misusing mfr info (unfortunately there are already some of these supported in HA).

Reply:
Oh nice. How did you find that out? :smile: 
If I swap to this, should I d,rop the ID or keep this as well?

Reply:
Google lol. It gets you other devs who've built other airthings things, their blog posts, their repos.

Reply:
I'd do both for now but that's just my instinct

Reply:
Lol! Yeah found a spec now myself. Will look into it a bit :)